Tribler/Test/Community/DHT/test_routing.py

```diff
@@ -0,0 +1,218 @@
+
+    def test_last_contact(self):
+        self.node.last_query = 5
+        self.assertEqual(self.node.last_contact, 5)
+
+        self.node.last_response = 10
+        self.assertEqual(self.node.last_contact, 10)
+
+
+class TestBucket(unittest.TestCase):
+
+    def setUp(self):
+        self.bucket = Bucket('01', max_size=8)
+
+    def test_owns(self):
+        pad_and_convert = lambda b: '{:<040X}'.format(int(b.ljust(160, '0'), 2)).decode('hex')
+        self.assertTrue(self.bucket.owns(pad_and_convert('01')))
+        self.assertTrue(self.bucket.owns(pad_and_convert('010')))
+        self.assertFalse(self.bucket.owns(pad_and_convert('00')))
+        self.assertFalse(self.bucket.owns(pad_and_convert('11')))
+
+    def test_get(self):
+        self.bucket.nodes[1]= 'node'
+        self.assertEqual(self.bucket.get(1), 'node')
+        self.assertEqual(self.bucket.get(2), None)
+
+    def test_add_(self):
+        node = FakeNode('01')
+        self.assertTrue(self.bucket.add(node))
+        self.assertEqual(self.bucket.get(node.id), node)
+
+    def test_add_fail(self):
+        node = FakeNode('11')
+        self.assertFalse(self.bucket.add(node))
+        self.assertEqual(self.bucket.get(node.id), None)
+
+    def test_add_update(self):
+        node1 = FakeNode('01')
+        node2 = FakeNode('01')
+        node2.address = ('1.2.3.4', 1)
+        self.assertEqual(self.bucket.last_changed, 0)
+        self.bucket.add(node1)
+        self.bucket.add(node2)
+        self.assertEqual(self.bucket.get(node1.id), node1)
+        self.assertEqual(self.bucket.get(node1.id).address, ('1.2.3.4', 1))
+        self.assertNotEqual(self.bucket.last_changed, 0)
+
+    def test_add_full(self):
+        self.bucket.max_size = 1
+        node1 = FakeNode('011')
+        node2 = FakeNode('010')
+        self.bucket.add(node1)
+        self.bucket.add(node2)
+        self.assertEqual(self.bucket.get(node1.id), node1)
+        self.assertEqual(self.bucket.get(node2.id), None)
+
+    def test_add_cleanup(self):
+        self.bucket.max_size = 1
+        node1 = FakeNode('011')
+        node2 = FakeNode('010')
+        self.bucket.add(node1)
+        node1.failed += 3
+        self.bucket.add(node2)
+        self.assertEqual(self.bucket.get(node1.id), node1)
+        self.assertEqual(self.bucket.get(node2.id), None)
+
+    def test_split(self):
+        self.bucket.max_size = 1
+        node = FakeNode('011')
+        self.bucket.add(node)
+
+        b010, b011 = self.bucket.split()
+        self.assertIsInstance(b010, Bucket)
+        self.assertIsInstance(b011, Bucket)
+        self.assertEqual(b010.get(node.id), None)
+        self.assertEqual(b010.prefix_id, '010')
+        self.assertEqual(b011.get(node.id), node)
+        self.assertEqual(b011.prefix_id, '011')
+
+    def test_split_not_full(self):
+        self.assertFalse(self.bucket.split())
+
+
+class TestRoutingTable(unittest.TestCase):
+    """
+    Class to test the routing table
```

Comment:
Please be consistent and either add a docstring under every class definition or not 👍